frontend/src/views/consumer/event-listing/EventBooking.jsx

```diff
@@ -0,0 +1,283 @@
+
+// posed components
+const Collapsible = posed.div({
+  collapsed: {
+    opacity: 0,
+    height: 0,
+    flip: true,
+  },
+  natural: {
+    opacity: 1,
+    height: '100%',
+    flip: true,
+  },
+});
+// styled componnents
+const Wrapper = styled.div`
+  padding: 1rem;
+  margin-top: 1rem;
+  width: 100%;
+  background-color: ${props =>
+    toRgba(
+      chroma(props.bgColor)
+        .alpha(0.3)
+        .rgba(),
+    )};
+`;
+const FormListItemWrapper = styled.div`
+  width: 100%;
+  margin-bottom: 0.5rem;
+  & > div {
+    padding: 0.5rem;
+    display: flex;
+    flex-direction: column;
+    overflow: hidden;
+
+    & > * {
+      margin: 0.5rem 0;
+    }
+    & form {
+      & > * {
+        margin-top: 0.5rem;
+      }
+    }
+  }
+`;
+const CustomerGroupButton = styled(Button)`
+  &&& {
+    padding: 0.5rem;
+    width: 100%;
+    height: auto;
+    font-size: 0.86rem;
+    border-radius: 8px;
+  }
+`;
+const FlexBoxHorizontal = styled(Collapsible)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 0.5rem 0;
+
+  & > *:first-child {
+    margin-right: 1rem;
+  }
+`;
+const TotalAmountCalculated = styled(FlexBoxHorizontal)`
+  background-color: rgba(0, 0, 0, 0.1);
+  padding: 0.5rem 1rem;
+  border-radius: 8px;
+  justify-content: space-between;
+  overflow: hidden;
+  * {
+    margin-bottom: 0;
+  }
+`;
+
+// child components
+const FormListItem = props => {
+  return (
+    <FormListItemWrapper>
+      <Typography type="largebody">{props.title}</Typography>
+      <Collapsible pose={props.disabled ? 'collapsed' : 'natural'}>
+        {props.children}
+      </Collapsible>
+    </FormListItemWrapper>
+  );
+};
+
+// main class
+export default connect('store')(
+  class EventBooking extends Component {
+    internalState = observable({
+      customerGroup: undefined,
+      tickets: [
+        {
+          id: 'first',
+          label: this.props.event.ticketCatalog[0].ticketDescription,
+          value: this.props.event.ticketCatalog[0].id,
+          amount: 0,
+        },
+      ],
+      phoneNumber: '',
+      name: '',
+      email: '',
+      classRoom: '',
+      school: '',
+    });
+    getTotalPrice = () => {
+      return this.internalState.tickets.reduce((acc, curr) => {
+        const targetCatalog = this.props.event.ticketCatalog.find(
+          elem => elem.id == curr.value,
+        );
+        if (targetCatalog) {
+          const singlePrice = targetCatalog.price;
+          const totalPriceForThisCatalog = singlePrice * curr.amount;
+          acc += totalPriceForThisCatalog;
+        }
+        return acc;
+      }, 0);
+    };
+    render() {
+      const { event } = this.props;
+      const { internalState } = this;
+
+      const isPrivateCustomer = internalState.customerGroup == 'private';
+      const isGroupConductorCustomer = internalState.customerGroup == 'group';
+      const totalCost = this.getTotalPrice();
+      const submittable = isGroupConductorCustomer
```

Comment:
How about making this a method of the component to make the logic easily testable?